### ISSUE TITLE
Downloaded files are semi-unplayable

### ISSUE THREAD

**Issue by forthrin** [spaam/svtplay-dl#601]
Files downloaded from viafree.no are semi-unplayable in VLC (video is tinted gray when skipping forward). Doing a manual conversion works around the issue:

    ffmpeg -i in.ts -codec copy out.mp4

Using the `--remux` option produces a file that is not playable at all (and the output has broken printf statements `Muxing %s into %s`). Don't know if this option is supposed to do the same as the above?

It would be nice to have this fixed, but may I suggest that this project is merged into `youtube-dl` which already supports the same functionality and sites (but where subtitle support is currently broken)?

**Comment by spaam** [comment]
does this happen to all videos downloaded from viafree.no?
remux will do `ffmpeg -i  orignal.ts -map 0:v -map 0:a -c copy -copyts -f mp4 -bsf:a aac_adtstoasc -y output.mp4`

**Comment by forthrin** [comment]
The following downloads a video which is tinted gray when skipping forward in VLC:

    svtplay-dl -S https://www.viafree.no/programmer/underholdning/paradise-hotel/sesong-8/episode-1

Using your remix snippet produces an unplayable file in VLC:

    ffmpeg -i original.ts -map 0:v -map 0:a -c copy -copyts -f mp4 -bsf:a aac_adtstoasc -y output.mp4

Removing `-copyts` from your snippet produces a playable file.

**Comment by spaam** [comment]
great. thanks :)

**Comment by forthrin** [comment]
Np. Why does this tool exists when `youtube-dl` exists? `svtplay-dl` lacks useful functions like playlist start index and filename formatting, so a project merge would seem like an energy saver for everyone.

**Comment by spaam** [comment]
why do we have debian, ubuntu, macos and windows? they all have somewhat same functionality?
why do we have different type of cars and bikes? they all do the same thing?

i think its good to have many things that do the same thing. competition is a good thing.
svtplay-dl have some features that youtube-dl dont have. like `-A` for example. download all videos for some site..  and no. i will not merge this project into youtube-dl.

**Comment by forthrin** [comment]
It's fun to have your own project, but when the benefit is marginal, well... Let's drop that debate :) Anyway, the following things are a minus at the moment.

1.  Filename formatting, so you can get a filename like `Paradise Hotel S08E10.mp4` rather than `paradise-hotel.s08e10-719140-viafree.ts´.
2. Improved resuming. Currently a download cannot be resumed, and resuming a playlist takes a long time to skip already downloaded files. So either quicker skipping or a manual `--playlist-start`.

**Comment by vagnretur** [comment]
I use Handbrake to convert all my files

**Comment by iwconfig** [comment]
Just a note here, correct me if i'm wrong but i think youtube-dl is in about the same age as svtplay-dl and i guess youtube-dl also was a fun little project once upon a time. I guess, as the benefit is marginal, whether svtplay-dl is merged into youtube-dl or not, it is not really an issue. Filename formatting is on the todo list, to do when there is time for it.

Oh btw, SVT, SVTplay and Öppet arkiv is supported by youtube-dl, but isn't viafree also supported?

**Comment by forthrin** [comment]
I switched to `svtplay-dl` because `youtube-dl` does not download subtitles for ViaFree (posted an issue with them for this). But then I'm running into other problems with `svtplay-dl`.

**Comment by iwconfig** [comment]
you could just download the subtitle only using `-S --force-subtitle`

**Comment by forthrin** [comment]
Yes, but the filenames are different, so then I need a script/alias to fix that.

**Comment by iwconfig** [comment]
Seems like a really easy fix with youtube-dl's `--exec`

**Comment by selepo** [comment]
@forthrin : Can you give example on any URL with same issue that isn't geoblocked ?
@spaam: Can the original issue be related to issue [#665](https://github.com/spaam/svtplay-dl/issues/665)

**Comment by forthrin** [comment]
I can't confirm it's a duplicate, but it's quite likely.

**Comment by forthrin** [comment]
@spaam: When will this be resolved? `--remux` still produces a semi-unplayable file (starts slowly or freezes when seeking). Last I checked, removing `-copyts` from the command line resolves the issue.

**Comment by selepo** [comment]
The PR (686) that I created will run remux on every HLS download. If we need to run another ffmpeg command, it should be quite easy to fix. But we would need to create a new subroutine in postprocess, and call that command instead of remux...

**Comment by granbom** [comment]
As I recall -copyts do a bit different things depending of it's placed before the input files or after.  Right now I think it's placed after the input file, maybe try before.

**Comment by forthrin** [comment]
I've never used `-copyts` for anything. `ffmpeg -i in.ts -codec copy out.mp4` works perfectly.
